### PR TITLE
[CELEBORN-183] Add replication for mappartition

### PR DESCRIPTION
**Pull request by zhongqiangczq**
### Does this PR introduce _any_ user-facing change?
@waitinfuture @RexXiong @FMX

### CODE REVIEW

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala [hunk 1]
```diff
@@ -58,8 +58,10 @@ class PushDataHandler extends BaseMessageHandler with Logging {
   var storageManager: StorageManager = _
   var conf: CelebornConf = _
   @volatile var pushDataTimeoutTested = false
+  var _worker: Worker = _
```
Comment (waitinfuture) [inline worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala:61]: We should not add Worker as PushHandler's member

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala [hunk 2]
```diff
@@ -755,6 +756,90 @@ class PushDataHandler extends BaseMessageHandler with Logging {
     }
   }
 
+  def replicateMapPartition(
+      shuffleKey: String,
+      location: PartitionLocation,
+      messageType: Type,
+      message: Message,
+      wrappedCallback: WrappedRpcResponseCallback): Unit = {
+    _worker.getMapPartitionReplicatePool(
```
Comment (waitinfuture) [inline worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala:765]: For ReducePartition, replicate pool is defined in PushHandler, why not keep consistent for MapPartition?

**Review by waitinfuture — APPROVED** [review]

**Comment by waitinfuture** [comment]
I think it's about time to refactor PushHandler.

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala (hunk 1 above)
Comment (zhongqiangczq) [reply worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala]: it's removed

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala (hunk 2 above)
Comment (zhongqiangczq) [reply worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala]: it defines a function val MapPartitionReplicatePool to keep consistent for reducePartition

**Comment by waitinfuture** [comment]
Let's hold this for sometime.

**Pull request closed without merging**